### PR TITLE
HPCC-18581 Upgrade codemirror

### PR DESCRIPTION
**Pull request by GordonSmith**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by GordonSmith** [comment]
@miguelvazq Please review (once https://github.com/hpcc-systems/HPCC-Platform/pull/10492 has been pulled)

**Comment by GordonSmith** [comment]
@miguelvazq this is ready for review now.

**Comment by GordonSmith** [comment]
@miguelvazq this has been rebased and is ready for review.

File: esp/src/eclwatch/dojoConfig.js [hunk 1]
```diff
@@ -32,8 +32,10 @@ function getConfig(env) {
             "@hpcc-js/map": baseUrl + "/node_modules/@hpcc-js/map/dist/map",
             "@hpcc-js/other": baseUrl + "/node_modules/@hpcc-js/other/dist/other",
             "clipboard": baseUrl + "/node_modules/clipboard/dist/clipboard",
+            "codemirror": baseUrl + "/node_modules/codemirror",
+            "crossfilter": baseUrl + "/crossfilter/crossfilter.min",
             "font-awesome": baseUrl + "/node_modules/@hpcc-js/common/font-awesome",
-            "crossfilter": baseUrl + "/crossfilter/crossfilter.min"
+            "tslib": baseUrl + "/node_modules/tslib/tslib"
```
Comment (miguelvazq) [inline esp/src/eclwatch/dojoConfig.js]: This code mirror related?

**Review by miguelvazq — COMMENTED** [review]
One minor question. FYI, also failed smoke test.

File: esp/src/eclwatch/dojoConfig.js (hunk 1 above)
Comment (GordonSmith) [reply esp/src/eclwatch/dojoConfig.js]: tslib is needed by CodeMirror.ts specifically its a couple of helper functions for TypeScript classes and its getting added now, as its the first TS class we are adding (there will be more)..

**Comment by miguelvazq** [comment]
Looks good. If smoke test fail is unrelated.

**Pull request merged**